channeldb/migration_11_invoices_test.go

```diff
@@ -0,0 +1,227 @@
+package channeldb
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec"
+	bitcoinCfg "github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/coreos/bbolt"
+	"github.com/lightningnetwork/lnd/zpay32"
+	litecoinCfg "github.com/ltcsuite/ltcd/chaincfg"
+)
+
+var (
+	testPrivKeyBytes = []byte{
+		0x2b, 0xd8, 0x06, 0xc9, 0x7f, 0x0e, 0x00, 0xaf,
+		0x1a, 0x1f, 0xc3, 0x32, 0x8f, 0xa7, 0x63, 0xa9,
+		0x26, 0x97, 0x23, 0xc8, 0xdb, 0x8f, 0xac, 0x4f,
+		0x93, 0xaf, 0x71, 0xdb, 0x18, 0x6d, 0x6e, 0x90,
+	}
+
+	testCltvDelta = int32(50)
+)
+
+// TestMigrateInvoices checks that invoices are migrated correctly.
+func TestMigrateInvoices(t *testing.T) {
+	t.Parallel()
+
+	payReqBtc, err := getPayReq(&bitcoinCfg.MainNetParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var ltcNetParams bitcoinCfg.Params
+	applyLitecoinParams(&ltcNetParams, &litecoinCfg.MainNetParams)
```

Comment:
The only thing we need from the net params for invoice creation/decoding is the `Bech32HRPSegwit` field. As a result, we don't need this entire function and can instead swap out this dedicated value.

Reply:
Ok, that's better. Changed.